addons/binding/org.openhab.binding.hs110/src/main/java/org/openhab/binding/hs110/handler/HS110Handler.java

```diff
@@ -0,0 +1,225 @@
+        refreshJob.cancel(true);
+    }
+
+    @Override
+    public void initialize() {
+        Configuration config = getThing().getConfiguration();
+
+        String ip = (String) config.get(CONFIG_IP);
+        logger.info("Initializing plug on ip {}", ip);
+        plug = new HS110(ip);
+
+        if (config.containsKey(CONFIG_REFRESH)) {
+            refresh = (BigDecimal) config.get(CONFIG_REFRESH);
+        } else {
+            refresh = new BigDecimal(REFRESH_DEFAULT);
+        }
+
+        startAutomaticRefresh();
+
+        updateStatus(ThingStatus.ONLINE);
+
+    }
+
+    private void startAutomaticRefresh() {
+        Runnable runnable = new Runnable() {
+            @Override
+            public void run() {
+                logger.debug("Refreshing {}", thing.getUID().getAsString());
+                try {
+                    boolean success = updateData();
+                    if (success) {
+                        updateState(new ChannelUID(getThing().getUID(), CHANNEL_SWITCH), getState());
+                        updateState(new ChannelUID(getThing().getUID(), CHANNEL_WATTAGE), getWattage());
+                        updateState(new ChannelUID(getThing().getUID(), CHANNEL_TOTAL), getTotal());
+                        updateState(new ChannelUID(getThing().getUID(), CHANNEL_SYSINFO), getSysinfo());
+                    }
+                } catch (Exception e) {
+                    logger.debug("Exception occurred during execution", e);
+                }
+            }
+        };
+
+        refreshJob = scheduler.scheduleAtFixedRate(runnable, 0, refresh.intValue(), TimeUnit.SECONDS);
+    }
+
+    private synchronized boolean updateData() {
+        logger.trace("Updating data for Plug type {}", thing.getThingTypeUID().getAsString());
+        try {
+            sysinfoData = plug.sendCommand(HS110.Command.SYSINFO);
+            if (thing.getThingTypeUID().equals(THING_TYPE_HS110)) {
+                energyData = plug.sendCommand(HS110.Command.ENERGY);
+                logger.debug("Updated energy Data for {}: {}", thing.getUID(), energyData);
+            }
+            logger.debug("Updated sysinfo Data for {}: {}", thing.getUID(), sysinfoData);
+            return true;
+
+        } catch (IOException e) {
+            logger.warn("Error accessing plug ", e);
+            energyData = null;
+            sysinfoData = null;
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, e.getMessage());
+            return false;
+        }
+    }
+
+    @Override
+    public Collection<ConfigStatusMessage> getConfigStatus() {
+
+        Collection<ConfigStatusMessage> configStatus = new ArrayList<>();
+
+        try {
+            InetAddress ip = InetAddress.getByName(plug.ip);
+            if (!ip.isReachable(500)) {
+                configStatus.add(
+                        ConfigStatusMessage.Builder.error("offline").withMessageKeySuffix("ip unreachable").build());
```

Comment:
`withMessageKeySuffix` sounds like this should be a reference to a valid key in the translation file?